### PR TITLE
assertHasSizeBetween works with strings. Fixes #1586

### LINKED ISSUE #1586: The assertion assertThat().HasSizeBetween() does not work with strings
#### Summary

The assertion assertThat().hasSizeBetween() with strings does not work, assertj-core seems to confuse string lengths between actual, min and max.

#### Example

```java
assertThat("Test1").hasSizeBetween(4, 11);
```
throws a confusing exception :
java.lang.IllegalArgumentException: The higher boundary <4> must be greater than the lower boundary <5>.

Tested on assertj_core 3.13.2 and AdoptOpenJDK 11

### PR DESCRIPTION
**Pull request by geoffreyarthaud**
#### Check List:
* Fixes #1586
* Unit tests : YES
* Javadoc with a code example (API only) : NA

### CODE REVIEW

File: src/test/java/org/assertj/core/internal/strings/Strings_assertHasSizeBetween_Test.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+public class Strings_assertHasSizeBetween_Test extends StringsBaseTest {
+
+  @Test
+  public void should_fail_if_actual_is_null() {
+    assertThatExceptionOfType(AssertionError.class).isThrownBy(() -> strings.assertHasSizeBetween(someInfo(), null, 2, 7))
+                                                   .withMessage(actualIsNull());
+  }
+
+  @Test
+  public void should_fail_if_size_of_actual_is_less_to_min_expected_size() {
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/internal/strings/Strings_assertHasSizeBetween_Test.java]: use GIVEN WHEN THEN

File: src/test/java/org/assertj/core/internal/strings/Strings_assertHasSizeBetween_Test.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    AssertionInfo info = someInfo();
+    String actual = "Han";
+
+    String errorMessage = shouldHaveSizeBetween(actual, actual.length(), 4, 7).create();
+    assertThatExceptionOfType(AssertionError.class).isThrownBy(() -> strings.assertHasSizeBetween(info, actual, 4, 7))
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/internal/strings/Strings_assertHasSizeBetween_Test.java]: use `AssertionsUtil.expectAssertionError` which makes the code a bit lighter, see https://github.com/joel-costigliola/assertj-core/blob/83cd126d183cf55fd3a16cc40c0f5f634ed71dc5/src/test/java/org/assertj/core/api/localdatetime/LocalDateTimeAssert_isCloseToUtcNow_Test.java#L42 as an example

File: src/test/java/org/assertj/core/internal/strings/Strings_assertHasSizeBetween_Test.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                                                   .withMessage(errorMessage);
+  }
+
+  @Test
+  public void should_fail_if_size_of_actual_is_greater_than_max_expected_size() {
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/internal/strings/Strings_assertHasSizeBetween_Test.java]: same comments as in the previous test

**Comment by joel-costigliola** [comment]
@geoffreyarthaud thanks for the PR! I have finished commenting, mainly cosmetic changes

**Comment by geoffreyarthaud** [comment]
Thanks for your review. I've updated the code, I hope I've understood your needs ! :sweat_smile:

**Comment by joel-costigliola** [comment]
Integrated thanks @geoffreyarthaud !

**Pull request merged**